linera-core/src/client.rs

```diff
@@ -1812,6 +1852,39 @@ where
         })
     }
 
+    /// Publishes some blob.
+    pub async fn publish_blob(
+        &mut self,
+        blob: Blob,
+    ) -> Result<ClientOutcome<(BlobId, Certificate)>, ChainClientError> {
+        let blob_id = BlobId::new(&blob);
+        self.node_client.cache_recent_blob(&blob_id, &blob).await;
+        self.add_pending_blob_to_chain_manager(&blob_id, &blob)
+            .await?;
```

Comment:
I think it would be cleaner to keep our local chain manager and storage state as similar as possible to that of a worker handling the proposal.

I'd rather put the blob into some temporary (optional) field local to the `ChainClient`, or pass it into the methods where it's needed. (Or maybe add it to local storage explicitly? But then we wouldn't clean it up if the proposal doesn't get finalized.)

Reply:
Or a vector, yes.

But independent of this, we will still need to keep blobs in the chain manager, too, but only if they are pending _from the point of view of the chain manager_, i.e. they are required by a block that we validated and that could still become confirmed.

Specifically, what we must prevent in the chain manager is a situation where a confirmed block gets signed by a quorum of validators but none of them have the blob anymore that that block requires.